fish_prompt.fish

```diff
@@ -847,7 +847,19 @@ function __bobthefish_prompt_rubies -S -d 'Display current Ruby information'
 end
 
 function __bobthefish_virtualenv_python_version -S -d 'Get current Python version'
-    switch (python --version 2>&1 | tr '\n' ' ')
+    set -l python_version
+
+    # try preferred Python version first
+    if [ (which python) ]
```

Comment:
As a general rule, any `()` in a fish prompt is a performance regression. Let's use the `type` builtin (specifically `type -fq python`) rather than `which` here.